processor/memorylimiterprocessor/factory.go

```diff
@@ -30,14 +31,22 @@ const (
 
 var processorCapabilities = consumer.Capabilities{MutatesData: false}
 
+type factory struct {
+	memoryLimiters map[config.ComponentID]*memoryLimiter
```

Comment:
Add comments to explain what the purpose of the map is.

Reply:
👍